core/trino-main/src/main/java/io/trino/execution/scheduler/SourcePartitionedScheduler.java

```diff
@@ -237,6 +237,12 @@ public synchronized void startLifespan(Lifespan lifespan, ConnectorPartitionHand
         scheduleGroups.put(lifespan, new ScheduleGroup(partitionHandle));
         whenFinishedOrNewLifespanAdded.set(null);
         whenFinishedOrNewLifespanAdded = SettableFuture.create();
+
+        // schedule a task for collecting dynamic filters in case some dependent split generator is waiting for them
+        if (dynamicFilterService.isCollectingTaskNeeded(stageExecution.getStageId().getQueryId(), stageExecution.getFragment())) {
+            checkArgument(lifespan.isTaskWide(), "Dynamic filtering cannot be used with grouped execution");
+            createTaskOnRandomNode();
```

Comment:
The old code as adding this task to `overallNewTasks`.  Do we not need to track this anymore?

Reply:
no, `io.trino.execution.scheduler.ScheduleResult#getNewTasks` is only used in tests.